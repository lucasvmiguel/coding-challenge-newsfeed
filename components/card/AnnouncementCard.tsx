import Card, { Icon, Columns, ColumnLeft, ColumnRight } from './Card'
import Markdown from '../Markdown'

const ICON_URL = 'https://avatars.dicebear.com/api/initials/a.svg'

export type Announcement = {
  id: number;
  title: string;
  body: string;
}

type Props = {
  announcement: Announcement;
}

export default function AnnouncementCard({ announcement }: Props) {
  return (
    <Card>
      <Columns>
        <ColumnLeft>
          <Icon src={ICON_URL} />
        </ColumnLeft>
        <ColumnRight>
          <h2 data-test="announcement-card-title">{announcement.title}</h2>
          <Markdown>{announcement.body}</Markdown>
        </ColumnRight>
      </Columns>
    </Card>
  )
}
