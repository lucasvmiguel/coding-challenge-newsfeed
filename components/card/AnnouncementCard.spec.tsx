import renderer from 'react-test-renderer'

import { announcement } from '../../test/mocks/announcement'

import AnnouncementCard from './AnnouncementCard'


it('renders AnnouncementCard component correctly - snapshot', () => {
  const tree = renderer
    .create(<AnnouncementCard announcement={announcement} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})