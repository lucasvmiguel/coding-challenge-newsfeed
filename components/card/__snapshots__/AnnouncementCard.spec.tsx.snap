// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AnnouncementCard component correctly - snapshot 1`] = `
<div
  className="sc-bdfBQB eDCjUP"
>
  <div
    className="sc-dlfnuX iOZAXF"
  >
    <div
      className="sc-hKgJUU jkblDv"
    >
      <img
        className="sc-gsTEea iRQMQw"
        src="https://avatars.dicebear.com/api/initials/a.svg"
      />
    </div>
    <div
      className="sc-eCstlR evEecB"
    >
      <h2
        data-test="announcement-card-title"
      >
        title testing
      </h2>
      <div
        className="sc-jSgvzq"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>body testing</p>
",
          }
        }
      />
    </div>
  </div>
</div>
`;
